app/views/spree/components/search/_products_search.html.erb

```diff
@@ -7,14 +7,14 @@
   selected = @taxon ? @taxon.id : params[:taxon]
 
   # Classes
-  class_names = [base_class]
-  class_names = class_names.join(" ")
+  classes = local_assigns.fetch(:classes, [])
+  class_names = classes.push(base_class).join(" ")
 %>
 
 <%= form_tag spree.products_path, class: class_names, method: :get do %>
   <% cache [I18n.locale, @taxons] do %>
     <%= render(
-      "spree/components/forms/inputs/select",
+      "spree/components/forms/inputs/select_input",
```

Comment:
I think that is better to move this change in a different commit. `Fix passing the local classes in partials` doesn't describe why this change is required.